### PR TITLE
Refactoring: extract floors configuration processing

### PR DESCRIPTION
**Pull request by CTMBNara**

### CODE REVIEW

File: src/main/java/org/prebid/server/util/algorithms/random/RandomAnyWeightedEntrySupplier.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        long reducedEntryNumber = entryNumber;
+        for (E entry : entries) {
+            reducedEntryNumber -= weight(entry);
+
+            if (reducedEntryNumber < 0) {
+                return entry;
+            }
+        }
+
+        throw new AssertionError();
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/util/algorithms/random/RandomAnyWeightedEntrySupplier.java:58]: Pls add comment like `// Should never happen`

File: src/main/java/org/prebid/server/util/algorithms/random/RandomPositiveWeightedEntrySupplier.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        super(weightExtractor);
+    }
+
+    protected int weight(E entry) {
+        final int weight = weightExtractor.apply(entry);
+        if (weight > 0) {
+            return weight;
+        }
+
+        throw new IllegalArgumentException("Non-positive weight.");
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/util/algorithms/random/RandomPositiveWeightedEntrySupplier.java:17]: Better 'Model weight should be greater then 0'

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/util/algorithms/random/RandomPositiveWeightedEntrySupplier.java (hunk 2 above)
Comment (CTMBNara) [reply src/main/java/org/prebid/server/util/algorithms/random/RandomPositiveWeightedEntrySupplier.java:17]: What is the "model" in this context? XD

Comment (SerhiiNahornyi) [reply src/main/java/org/prebid/server/util/algorithms/random/RandomPositiveWeightedEntrySupplier.java:17]: than `Wweight should be greater then 0` )

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**